invenio_rdm_records/services/communities/service.py

```diff
@@ -325,3 +325,26 @@ def search_suggested_communities(
             extra_filter=communities_filter,
             **kwargs,
         )
+
+    @unit_of_work()
+    def set_default(self, identity, id_, data, uow):
+        """Set default community."""
+        record = self.record_cls.pid.resolve(id_)
+
+        self.require_permission(identity, "manage", record=record)
+
+        errors = []
+        try:
+            record.parent.communities.default = data["data"]
```

Comment:
Major: Usually when processing data, we do use marshmallow schemas to validate it before processing.